### PR TITLE
Code coverage: Transaction service - Subtask-3

### PR DESCRIPTION
**Pull request by Dimple16**
### Changes made
- Added missing models
- Added param and return types for all methods

### CODE REVIEW

File: src/app/core/models/sort-filters-params.model.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { FilterQueryParams } from './filter-query-params.model';
+
+export interface SortFiltersParams {
+  pageNumber: number;
+  queryParams: FilterQueryParams;
+  sortParam: string;
+  sortDir: string;
```
Comment (rvab) [inline src/app/core/models/sort-filters-params.model.ts:7]: What will be the values here?. If it's `desc` and `asc`, then you can use `sortDir: 'asc' | 'desc'`. This helps in preventing passing random string

File: src/app/core/services/transaction.service.ts [hunk 2]
```diff
@@ -42,6 +50,15 @@ type PaymentMode = {
   key: string;
 };
 
+type PaymentModeSummary = {
```
Comment (rvab) [inline src/app/core/services/transaction.service.ts:53]: We are creating models/types in different files, no? Any reason for keeping it here?

File: src/app/core/services/transaction.service.ts [hunk 3]
```diff
@@ -136,12 +153,12 @@ export class TransactionService {
     cacheBusterObserver: transactionsCacheBuster$,
   })
   getMyExpenses(
-    config: Partial<{ offset: number; limit: number; order: string; queryParams: any }> = {
+    config: Partial<{ offset: number; limit: number; order: string; queryParams: {} }> = {
```
Comment (rvab) [inline src/app/core/services/transaction.service.ts:156]: * what is the type of `queryParams`? If we can't define model for it, let's use [`Record`](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type)

File: src/app/core/services/transaction.service.ts [hunk 4]
```diff
@@ -174,7 +191,7 @@ export class TransactionService {
   @Cacheable({
     cacheBusterObserver: transactionsCacheBuster$,
   })
-  getAllExpenses(config: Partial<{ order: string; queryParams: any }>) {
+  getAllExpenses(config: Partial<{ order: string; queryParams: {} }>): Observable<Expense[]> {
```
Comment (rvab) [inline src/app/core/services/transaction.service.ts:194]: same as above

File: src/app/core/services/transaction.service.ts [hunk 5]
```diff
@@ -189,14 +206,14 @@ export class TransactionService {
...
-      reduce((acc, curr) => acc.concat(curr), [] as any[])
+      reduce((acc, curr) => acc.concat(curr), [] as Expense[])
     );
   }
 
   @Cacheable({
     cacheBusterObserver: transactionsCacheBuster$,
   })
-  getTransactionStats(aggregates: string, queryParams = {}) {
+  getTransactionStats(aggregates: string, queryParams = {}): Observable<any> {
```
Comment (rvab) [inline src/app/core/services/transaction.service.ts:216]: * same comment as above for `queryParams`.
* Why use `any` here?

File: src/app/core/services/transaction.service.ts [hunk 6]
```diff
@@ -363,23 +380,23 @@ export class TransactionService {
     );
   }
 
-  getETxnc(params: { offset: number; limit: number; params: any }) {
+  getETxnc(params: { offset: number; limit: number; params: {} }): Observable<Expense[]> {
```
Comment (rvab) [inline src/app/core/services/transaction.service.ts:383]: same comment as above for queryParams

File: src/app/core/services/transaction.service.ts [hunk 7]
```diff
@@ -363,23 +380,23 @@ export class TransactionService {
...
     return this.apiV2Service
       .get('/expenses', {
         ...params,
       })
       .pipe(map((etxns) => etxns.data));
   }
 
-  getMyExpensesCount(queryParams = {}) {
+  getMyExpensesCount(queryParams = {}): Observable<number> {
```
Comment (rvab) [inline src/app/core/services/transaction.service.ts:391]: same as above

**Review by rvab — CHANGES_REQUESTED** [review]

File: src/app/core/services/transaction.service.ts (hunk 5 above)
Comment (Dimple16) [reply src/app/core/services/transaction.service.ts]: - fixed all the references where `queryParams` was used
- the usage of `getTransactionStats` needs to be fixed throughout, in other files as well, a lot of changes will get involved, added a `TODO` here

File: src/app/core/services/transaction.service.ts (hunk 2 above)
Comment (Dimple16) [reply src/app/core/services/transaction.service.ts]: this is moved to a different file

File: src/app/core/models/sort-filters-params.model.ts (hunk 1 above)
Comment (Dimple16) [reply src/app/core/models/sort-filters-params.model.ts:7]: the usage of `sortDir` in the sort method is different than how it is used in the Filters model, we will need to change it throughout, so using the same here, in few places an enum is used, it is incompatible here

**Review by rvab — APPROVED** [review]

**Pull request merged**